samtranslator/swagger/swagger.py

```diff
@@ -846,6 +850,8 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
         """
         if resource_policy is None:
             return
+        if not isinstance(resource_policy, dict):
+            raise InvalidDocumentException(["Invalid resource policy"])
```

Comment:
Make sure to provide some details about where this is in the template. We generally do this through adding the logical id, so the customer at least knows what resource they need to dig into. This goes for all the exceptions, not just this one.